src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/HttpRequestManager.cs

```diff
@@ -75,6 +75,16 @@ public async Task<HttpResponseMessage> ExecuteAsync(int retryCount = 3)
 
                             retriesRemaining = 0;
                         }
+                        else if (response.StatusCode == HttpStatusCode.Unauthorized)
```

Comment:
I wonder if we should instead default to not retry if the status code is in the 400s range, and add exceptions for
* 404
* 429 (Maybe not as part of this pr, but we can even look at the response headers when it's a 429, they usually include information for when you will stop being throttled) ?

Status codes in the 400s usually won't give you a different response no matter how many times you try again.

After all, if you receive a [418](https://en.wikipedia.org/wiki/Hyper_Text_Coffee_Pot_Control_Protocol) The server won't stop being a teapot no matter how many times you keep retrying.

Reply:
404 and 400 default to not retrying today and should probably stay that way. 404 is a really common case, for instance when you attempt to access the version.details.xml file in a sha where it doesn't exist.

Have we seen any cases where the followup for the 401 passed?

Reply:
Yeah, we probably can't enable them for 404s then, but I guess my point is, instead of having to add every code in the 400s that we don't want to retry, ( I can really only think of 429 as something we'd eventually want to retry due to throttling) we should invert it, and not retry anything in the 400s, and add exceptions for the ones we do want to retry.

If not, we should add both 401 and 403 from this PR. I don't think there's ever been a case where we get one of those and then succeed afterwards.

Reply:
IMO what you suggest is a fair optimization @riarenas . But perhaps would be better to just keep things simple and add the 403 to the "list". If the list start to grow we can refactor it?

Reply:
@riarenas - do you think this is a blocking concern?

Reply:
It Isn't, but at the least 403 shol d be added to the l ist in the PR.

Reply:
Done